### PR TITLE
Use react context to share state between components/pages

### PR DESCRIPTION
**Pull request by robin-thomas**

### CODE REVIEW

File: components/dashboard/DashboardBlocks.js [hunk 1]
```diff
@@ -43,69 +45,30 @@ const DashboardBlocks = () => (
...
-            </Link>
-          </Col>
-        </Row>
+        <DataConsumer>
+          {context => (
+            <Row className="gutter-4">
+              {context.blocks.map((block, index) => {
+                return (
+                  <DashboardBlocksBlock
+                    key={context.blocks[index].blockNumber}
```
Comment (frol) [inline components/dashboard/DashboardBlocks.js]: Can we use `block` instead of `context.block[index]`?

File: components/HeaderDropdownNav.js [hunk 2]
```diff
@@ -2,48 +2,37 @@ import { Component } from "react";
...
-            onClick={this.selectNetwork}
-          />
-          <HeaderDropdownItem
-            title="Testnet Two"
-            onClick={this.selectNetwork}
-          />
+          {context.networks.map((network, index) => {
+            return (
+              <HeaderDropdownItem
+                key={index}
```
Comment (frol) [inline components/HeaderDropdownNav.js]: I think the `network` string is a better key than `index`.

File: components/dashboard/DashboardBlocks.js (hunk 1 above)
Comment (robin-thomas) [reply components/dashboard/DashboardBlocks.js]: Yes. We can. My mistake. Will fix it now.

File: components/HeaderDropdownNav.js (hunk 2 above)
Comment (robin-thomas) [reply components/HeaderDropdownNav.js]: Updated.

File: components/dashboard/DashboardBlocks.js (hunk 1 above)
Comment (robin-thomas) [reply components/dashboard/DashboardBlocks.js]: Updated

File: components/HeaderDropdownNav.js (hunk 2 above)
Comment (frol) [reply components/HeaderDropdownNav.js]: @robin-thomas I see that you have changed the implementation of `updateNetwork`, but the `key` is still `index`.

**Pull request merged**